backends/dpdk/dpdkContext.cpp

```diff
@@ -236,16 +270,17 @@ DpdkContextGenerator::addMatchAttributes(const IR::P4Table*table, const cstring
         if (attr.params) {
             int index = 0;
             int position = 0;
+            int param_width = 8;   // Minimum width for dpdk action params
             for (auto param : *(attr.params)) {
-                // TODO Handle other types of parameters
                 if (param->type->is<IR::Type_Bits>()) {
-                    addImmediateField(immFldArray, param->name.originalName,
-                                      index/8, param->type->width_bits());
-                    index += param->type->width_bits();
-                    position++;
-                } else {
+                    param_width = param->type->width_bits();
+                } else if (!param->type->is<IR::Type_Boolean>()) {
                     BUG("Unsupported parameter type %1%", param->type);
                 }
```

Comment:
Earlier error was thrown for param type other than Type_Bits. Now should error be limited to only Type_Boolean ?

Reply:
No error for Type_Bits and Type_Boolean. Other types are unsupported

Reply:
Okay. got it. The minimum param width is 8 in dpdk for any param type. Looks fine